services/src/pro/contexts/mod.rs

```diff
@@ -24,3 +37,202 @@ pub trait ProContext: Context<Session = UserSession> {
     fn user_db_ref(&self) -> &Self::UserDB;
     fn oidc_request_db(&self) -> Option<&OidcRequestDb>;
 }
+
+pub struct ExecutionContextImpl<S, D, L>
+where
+    D: DatasetDb<S>,
+    L: LayerProviderDb,
+    S: Session,
+{
+    dataset_db: Arc<D>,
+    layer_provider_db: Arc<L>,
+    thread_pool: Arc<ThreadPool>,
+    session: S,
+    tiling_specification: TilingSpecification,
+}
+
+impl<S, D, L> ExecutionContextImpl<S, D, L>
+where
+    D: DatasetDb<S>,
+    L: LayerProviderDb,
+    S: Session,
+{
+    pub fn new(
+        dataset_db: Arc<D>,
+        layer_provider_db: Arc<L>,
+        thread_pool: Arc<ThreadPool>,
+        session: S,
+        tiling_specification: TilingSpecification,
+    ) -> Self {
+        Self {
+            dataset_db,
+            layer_provider_db,
+            thread_pool,
+            session,
+            tiling_specification,
+        }
+    }
+}
+
+impl<S, D, L> ExecutionContext for ExecutionContextImpl<S, D, L>
+where
+    D: DatasetDb<S>
+        + SessionMetaDataProvider<
+            S,
+            MockDatasetDataSourceLoadingInfo,
+            VectorResultDescriptor,
+            VectorQueryRectangle,
+        > + SessionMetaDataProvider<S, OgrSourceDataset, VectorResultDescriptor, VectorQueryRectangle>
+        + SessionMetaDataProvider<S, GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>,
+    L: LayerProviderDb,
+    S: Session,
+{
+    fn thread_pool(&self) -> &Arc<ThreadPool> {
+        &self.thread_pool
+    }
+
+    fn tiling_specification(&self) -> TilingSpecification {
+        self.tiling_specification
+    }
+
+    fn initialize_operator(
+        &self,
+        op: Box<dyn geoengine_operators::engine::InitializedRasterOperator>,
+        span: Box<dyn geoengine_operators::engine::CreateSpan>,
+    ) -> Box<dyn geoengine_operators::engine::InitializedRasterOperator> {
+        Box::new(InitializedProcessorStatistics::statistics_with_id(
+            op,
+            "gdal_source".to_string(),
```

Comment:
why is everything a gdal source?

Reply:
yeah this is wrong. I could add a `get_id` method to `CreateSpan` to set it here, or just get rid of the "id" here. It's somewhat redundant because it would always be the same for every span.

Reply:
I removed the id for now because it was exactly the same as the span, thus carried no information. 68617472345bdacfa1a856184f118722e110b5b3